packages/NodeNameResolver/NodeNameResolver.php

```diff
@@ -201,7 +198,15 @@ public function isStringName(string $resolvedName, string $desiredName): bool
             return $desiredName === $resolvedName;
         }
 
-        if (StringUtils::isMatch($desiredName, self::CONTAINS_WILDCARD_CHARS_REGEX)) {
+        $containsWildcard = false;
+        foreach(self::REGEX_WILDCARD_CHARS as $char) {
+            if (str_contains($desiredName, $char)) {
+                $containsWildcard = true;
+                break;
+            }
+        }
+
+        if ($containsWildcard) {
```

Comment:
When I see this method, I wonder if it should be doing regex check at all. It's a bit magical behavior from `isName()` and in 99.9 % cases never used.

Instead there should be `matchName()` method dedicated for this. I've run tests without it and it only reports ~10 fixtures.

Reply:
it would be great if we could reduce regex usage to only the necessary places.

after this change it no longer shows up in the memory focused profiles, but the method here is still the top1 method regarding CPU use - even after the PR

<img width="684" alt="grafik" src="https://github.com/rectorphp/rector-src/assets/120441/a04e5198-9c8d-4a07-835a-d96b138effcf">

Reply:
Could you give the separated method a try?

Reply:
if you could give me a hint how to get rector running again :)

https://github.com/rectorphp/rector/issues/8191